radis/spectrum/spectrum.py

```diff
@@ -3574,6 +3575,159 @@ def __str__(self):
+        :func:`~radis.spectrum.operations.Transmittance_noslit`  ::
+
+            Radiance(s).min() 
+        """
+
+        var = self._get_unique_var(operation_name="min")
+        w, I = self.get(var, wunit=self.get_waveunit(), copy=False)
+        return I.min()
+
+    def normalize(
+        self, normalize_how="max", wrange=(), wunit=None, inplace=False, force=False
+    ):
+        """ Normalise the Spectrum, if only one spectral quantity is available.
+        
+        Parameters
+        ----------
+                
+            normalize_how: ``'max'``, ``'area'``, ``'mean'``
+                how to normalize. ``'max'`` is the default but may not be suited for very
+                noisy experimental spectra. ``'area'`` will normalize the integral to 1.
+                ``'mean'`` will normalize by the mean amplitude value
+
+            wrange: tuple
+                if not empty, normalize on this range
+                
+            wunit: ``"nm"``, ``"cm-1"``, ``"nm_vac"``
+                unit of the normalisation range above. If ``None``, use the 
+                spectrum default waveunit. 
+                
+            inplace: bool
+                if ``True``, changes the Spectrum. 
+                
+        Other Parameters
+        ----------------
+        
+            force: bool
+                By default, normalizing some parametres such as transmittance 
+                is forbidden because considered non-physical. Use force=True
+                if you really want to. 
+        
+        Examples
+        --------
+        
+            s.normalize("max", (4200, 4800), inplace=True)
+        
+        """
+
+        from radis.spectrum.operations import multiply
+
+        var = self._get_unique_var(operation_name="normalize")
+
+        if var in ["transmittance", "transmittance_noslit"] and not force:
+            raise ValueError(
+                "Cannot normalize {0}. Use force=True if you really want.".format(var)
+            )
+
+        s = self
+
+        if wunit is None:
+            wunit = s.get_waveunit()
+
+        if wrange is not None and len(wrange) > 0:
+            wmin, wmax = wrange
+            w, I = s.get(var, wunit=wunit, copy=False)  # (faster not to copy)
+            b = (w > wmin) & (w < wmax)
+            if normalize_how == "max":
+                norm = np.nanmax(I[b])
+                norm_unit = s.units[var]
+            elif normalize_how == "mean":
+                norm = np.nanmean(I[b])
+                norm_unit = s.units[var]
+            elif normalize_how == "area":
+                norm = np.abs(nantrapz(I[b], w[b]))
+                norm_unit = u.Unit(s.units[var]) * u.Unit(wunit)
+            else:
+                raise ValueError(
+                    "Unexpected `normalize_how`: {0}".format(normalize_how)
+                )
+
+            out = multiply(s, 1 / norm, unit=norm_unit, inplace=inplace)
+
+        else:
```

Comment:
(details)
I would make clearer the if ... then ... else here


```
 if len(wrange) == 2:
             blabla
 elif isinstance(wrange, (int, float)):
             blabla
  else:
             UserWarning

```

Reply:
Great, and it would also to normalize at one specific point. I forgot this one :)